dpcontracts.py

```diff
@@ -546,6 +552,124 @@ def inner(*args, **kwargs):
         return inner
     return func
 
+def pytypes():
```

Comment:
If possible, could we name this just "types"? That way the user could do `import types from contracts` or just use `contracts.types`, which (IMHO) seems cleaner.